test_read_only_default/test_read_only_default.py

```diff
@@ -0,0 +1,60 @@
+from django.db import models
+from rest_framework import serializers
+from rest_framework.validators import (
+    UniqueTogetherValidator,
+)
+
+from rest_framework import viewsets
+import json
+from rest_framework import mixins
+from rest_framework import generics
+from rest_framework.views import APIView
+from rest_framework.response import Response
+from rest_framework import status
+from django.test import TestCase
+from rest_framework import status
+
+
+class ModelTwo(models.Model):
+    testfieldtwo = models.CharField(max_length=60, primary_key=True)
+
+class ModelOne(models.Model):
+    testfield = models.ForeignKey(ModelTwo, on_delete=models.CASCADE)
+    alias = models.CharField(max_length=60)
+    name = models.CharField(max_length=60)
+
+    class Meta:
+        unique_together = ('testfield', 'name',)
+
+
+class ExampleSerializer(serializers.ModelSerializer):
+    altername = serializers.CharField(source="testfield", default=ModelTwo(testfieldtwo="model 2 default"), read_only=True)
+    
+    class Meta:
+        model = ModelOne
+        fields = ('altername' ,'alias', 'name')
+        validators = [
+            UniqueTogetherValidator(
+                queryset=ModelOne.objects.all(),
+                fields=('testfield', 'name')
+            )
+        ]
+
+class ExampleViewSet(mixins.CreateModelMixin,
+                             mixins.UpdateModelMixin,
+                             viewsets.ReadOnlyModelViewSet):
+
+    queryset = ModelOne.objects.all()
+    serializer_class = ExampleSerializer
```

Comment:
I actually have a codebase that was throwing these errors. I am upgrading from drf 3.7.7 to drf 3.10.x. So i wrote these even same as the codebase so that i could emulate the exact errors. And i wrote these even before you asked me to write tests. So these are overly complicated and redundant.

Reply:
That makes sense - I assumed you were pulling your examples from existing work. Just pointing out how they can be simplified.